central/sensor/service/pipeline/nodeinventory/pipeline.go

```diff
@@ -56,49 +57,50 @@ func (p *pipelineImpl) Match(msg *central.MsgFromSensor) bool {
 func (p *pipelineImpl) Run(ctx context.Context, clusterID string, msg *central.MsgFromSensor, _ common.MessageInjector) error {
 	defer countMetrics.IncrementResourceProcessedCounter(pipeline.ActionToOperation(msg.GetEvent().GetAction()), metrics.NodeInventory)
 
+	// Sanitize input.
 	event := msg.GetEvent()
 	ninv := event.GetNodeInventory()
 	if ninv == nil {
 		return errors.Errorf("unexpected resource type %T for node inventory", event.GetResource())
 	}
-
-	log.Infof("Received NodeInventory for Node name='%s' ID='%s'", ninv.GetNodeName(), ninv.GetNodeId())
-	log.Debugf("NodeInventory for name='%s' contains %d packages to scan from %d content sets", ninv.GetNodeName(),
+	invStr := fmt.Sprintf("for node %s (id: %s)", ninv.GetNodeName(), ninv.GetNodeId())
+	log.Infof("received node inventory %s", invStr)
+	log.Debugf("node inventory %s contains %d packages to scan from %d content sets", invStr,
 		len(ninv.GetComponents().GetRhelComponents()), len(ninv.GetComponents().GetRhelContentSets()))
-
-	if event.GetAction() == central.ResourceAction_REMOVE_RESOURCE {
-		log.Warn("Deletion of NodeInventories is not supported")
+	if event.GetAction() != central.ResourceAction_UNSET_ACTION_RESOURCE {
+		log.Errorf("node inventory %s with unsupported action: %s", invStr, event.GetAction())
 		return nil
 	}
-
 	ninv = ninv.Clone()
 
-	// TODO(ROX-14484): Resolve the race between pipelines - Start of critical section
+	// Read the node from the database, if not found we fail.
 	node, found, err := p.nodeDatastore.GetNode(ctx, ninv.GetNodeId())
```

Comment:
Here, if you are pulling the node only for the metadata, then you can use `GetNodeMetadata`. It is more performant because no cycles are spent on pulling scan data.

Reply:
But if enriching fails, later on, I will overwrite with an empty scan. I leave it as is.

Reply:
I will follow up on this in a separate PR. I want to add a single API to retrieve one single NodeMetadata instead of using GetMany().

Reply:
Also need to double-check what metadata brings in and see if we are not missing any key piece of information. And write tests around it.

Reply:
Sounds like a reasonable follow-up action.